src/concurrent_server.py

```diff
@@ -0,0 +1,104 @@
+# _*_ coding: utf-8 _*_
+"""Create a server that can handle concurrent messages."""
+import os
+import io
+import mimetypes
+
+
+def server(socket, address):
+
+    msg = b''
+    buffer_length = 8
+    message_complete = False
+    while not message_complete:
+        data = socket.recv(buffer_length)
+        msg += data
+        if len(data) < buffer_length:
+            break
+    msg = msg.decode('utf-8')
+    try:
+        path = parse_request(msg)
+        get_stuff = resolve_uri(path)
+        response = response_ok(get_stuff)
+    except NameError:
+        response = response_error("405", "Method not allowed")
+    except AttributeError:
+        response = response_error("403", "Forbidden")
+    except LookupError:
+        response = response_error("400", "Bad Request")
+    except IOError:
+        response = response_error("404", "File Not Found")
+    socket.sendall(response)
+    socket.close()
+
+
+def parse_request(argument):
```

Comment:
Out of curiosity, is there any difference between the `parse_request`, `resolve_uri`, `response_ok`, and `response_error` functions in this file and those in the `server.py` module?  If not, why not import these functions from there to avoid re-typing them?  

The problem with having a second file with a bunch of repeated functions is that if you someday wish to refactor one of them, or fix a bug, you'll need to fix it in both places.  Importing allows you to avoid this.